### ISSUE TITLE
[i] 2.1.2 "No Keyboard Trap" - change widget from date picker to something else

### ISSUE THREAD

**Issue by nitedog** [w3c/wai-wcag-videos#65]
[Comment from Detlev in the survey](https://www.w3.org/2002/09/wbs/1/WCAGvideos_batch2thorough/results#xq6) and Rain separately:

> Paraphrased: Change the widget from date picker (because it has its own complexities) to something else, for example a script re-focussing a field when the next field is tabbed.

**Comment by MicheleAWilliams-A11y** [comment]
So, I think the key to consider is that a date picker _should_ in some ways be a keyboard trap -- like modal dialogs, focus should stay within the calendar until the person says they want to leave. I see, however, in this scenario that focus presumably jumps to the calendar automatically and then doesn't have any way of closing it or moving out. For those reasons, I think the date picker is a good example (that is, I vote to keep it). 

The interaction behavior does need to better align with typical, expected behavior, but overall I think the example would resonate with folks. Since a date picker is not a simple native component (e.g., use `<button>` instead of `<div>`) and thus you *have* to build a custom solution, it seems good to reinforce for folks that part of your development should include keyboard functionality -- such as allowing focus to move to or past it, and the ability to close it.

Here are some suggestions for how the date picker should work if kept:

**Bad example:** Focus jumps automatically to within the calendar when focus is moved to the date field. You can clarify that the subject first tries Escape because they want to type out their date rather than use the calendar, but that doesn't close it as is the usual behavior. So then they take the longer route to navigate around with Tab and arrow keys (which correctly moves within the calendar dialog such as to arrows, dropdowns, and dates) but when they press Enter or Space to make a selection nothing happens because it only works with a mouse click. They accidentally find that Backspace closes the dialog and puts the cursor in the field so they could type out the date, like they wanted to do in the beginning.

**Good example:** A "calendar icon" button is part of the focus order so the subject can type their entry or explicitly open the calendar and move focus rather than automatically having that happen. Then continue to point out there's a prominent "X" to close (which can be activated by Enter or Space), the date selections can be activated by Enter or Space, or the person can press "Escape". I would take out the on-screen instructions though -- that's not really realistic to have and the presence of an "X" to close is usually enough. Designers will likely baulk if they see bulky instructions on a small widget like a calendar.

**Comment by rainbreaw** [comment]
Michele's suggestion above addresses my concern.
